client/task-list/tasks/payments/methods.js

```diff
@@ -192,10 +192,16 @@ export function getPaymentMethods( {
 				! hasCbdIndustry,
 			plugins: [ 'woocommerce-gateway-stripe' ],
 			container: <Stripe />,
+			// Whether publishable and secret keys are filled for given mode.
 			isConfigured:
 				options.woocommerce_stripe_settings &&
-				options.woocommerce_stripe_settings.publishable_key &&
-				options.woocommerce_stripe_settings.secret_key,
+				( ( options.woocommerce_stripe_settings.testmode === 'no' &&
```

Comment:
The logic here is getting very convoluted and looks like it would be difficult to test and maintain. Can you split this out into it's elements above?